src/ext/ddtrace.c

```diff
@@ -349,19 +350,19 @@ static PHP_FUNCTION(dd_trace_dd_get_memory_limit) {
     PHP5_UNUSED(return_value_used, this_ptr, return_value_ptr, ht);
     PHP7_UNUSED(execute_data);
 
-    RETURN_LONG(get_memory_limit(TSRMLS_C));
+    RETURN_LONG(ddtrace_get_memory_limit(TSRMLS_C));
 }
 
 /* {{{ proto bool dd_trace_check_memory_under_limit() */
 static PHP_FUNCTION(dd_trace_check_memory_under_limit) {
     PHP5_UNUSED(return_value_used, this_ptr, return_value_ptr, ht);
     PHP7_UNUSED(execute_data);
 
-    static zend_long limit = -1;
+    static int64_t limit = -1;
```

Comment:
Is there a reason to avoid zend_long here? I think PHP publishes `Zend/zend_long.h`, so we can just use `zend_long` from PHP for our default integer types and we wouldn't have to worry about integer type mismatches at all.

Reply:
@morrisonlevi good point - looking at https://github.com/DataDog/dd-trace-php/pull/496/files#diff-044fd8e3f6c94a464fabd5a901bbcf44L43 it seems that we had to override it for compatibility with 5.x - having looked at 5.4 I can only confirm that only `zend_long64` is defined there.

Reply:
Prior to version 7, PHP used "long", which means it was only 32 bit on Windows but 64 bit in most other places. Here's the definition for a zval in PHP 5.4, for example: https://github.com/php/php-src/blob/PHP-5.4/Zend/zend.h#L307-L316.

Reply:
At the moment we don't support 32 bit - however casting from 32 bit to 64 bit on a signed integer should "still work"(tm)